<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EnumParameterView</class>
 <widget class="QFrame" name="EnumParameterView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>281</width>
    <height>173</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Frame</string>
  </property>
  <property name="frameShape">
   <enum>QFrame::StyledPanel</enum>
  </property>
  <property name="frameShadow">
   <enum>QFrame::Raised</enum>
  </property>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>0</y>
     <width>265</width>
     <height>155</height>
    </rect>
   </property>
   <property name="title">
    <string>Enumeration Parameter</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
   <widget class="QLineEdit" name="numberText">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>20</y>
      <width>50</width>
      <height>20</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="maximumSize">
     <size>
      <width>50</width>
      <height>50</height>
     </size>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="descriptionText">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>45</y>
      <width>133</width>
      <height>20</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>45</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Description</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>44</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Identifier</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="isWriteableBox">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>100</y>
      <width>81</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Is Writeable</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="isCommandCheckBox">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>120</y>
      <width>82</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Is Command</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>70</y>
      <width>25</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Value</string>
    </property>
   </widget>
   <widget class="QComboBox" name="valueComboBox">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>70</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="editItemsButton">
    <property name="geometry">
     <rect>
      <x>176</x>
      <y>70</y>
      <width>24</width>
      <height>22</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>24</width>
      <height>22</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>24</width>
      <height>22</height>
     </size>
    </property>
    <property name="toolTip">
     <string>Edit Enumeration</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <activeon>:/image/resources/ui-combo-box-edit.png</activeon>
     </iconset>
    </property>
   </widget>
   <widget class="QLineEdit" name="identifierText">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>20</y>
      <width>133</width>
      <height>20</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>descriptionText</sender>
   <signal>editingFinished()</signal>
   <receiver>EnumParameterView</receiver>
   <slot>updateEntityDescription()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>146</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isCommandCheckBox</sender>
   <signal>clicked()</signal>
   <receiver>EnumParameterView</receiver>
   <slot>updateEntityCommand()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isWriteableBox</sender>
   <signal>clicked()</signal>
   <receiver>EnumParameterView</receiver>
   <slot>updateEntityWriteable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>valueComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>EnumParameterView</receiver>
   <slot>updateEntityValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>130</x>
     <y>79</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateUi()</slot>
  <slot>updateEntityDescription()</slot>
  <slot>updateEntityWriteable()</slot>
  <slot>updateEntityCommand()</slot>
  <slot>updateEntityValue(int)</slot>
 </slots>
</ui>
